### ISSUE TITLE
CSGShape subtraction sometimes not visible

### ISSUE THREAD

**Issue by payload** [godotengine/godot#30601]
Please search existing issues for potential duplicates before filing yours:
Search for open CSGShape issues. Didn't found any matching topic.

**Godot version:** 3.1
**OS/device including version:** Ubuntu 18.10
**Issue description:**
A CSGBox subtracting from a larger CSGBox sometimes has no visible effect.
Can happen also with other CSGShapes.
Expected is that I can subtract always a CSGShape from another and when they overlap, it is indeed cut out correctly.
**Steps to reproduce:** [video](https://cdn.discordapp.com/attachments/279004126795268106/600410981365186571/godot-2019-07-15_21.34.50.webm)
**Minimal reproduction project:** Move the small box a little. Example scene.tres:
```
[gd_scene format=2]

[node name="Spatial" type="Spatial"]

[node name="CSGCombiner" type="CSGCombiner" parent="."]

[node name="CSGBox" type="CSGBox" parent="CSGCombiner"]
width = 100.0
depth = 100.0

[node name="CSGBox2" type="CSGBox" parent="CSGCombiner"]
operation = 2
snap = 0.1
width = 3.0
height = 3.0
depth = 3.0
```

**Comment by Calinou** [comment]
Does this still occur if you set the CSGBox2's `snap` property to a lower value (like the default)?

**Comment by payload** [comment]
> Does this still occur if you set the CSGBox2's `snap` property to a lower value (like the default)?

Yes. The snap value does not have an effect in these cases. The issue persists.
Sorry I played around with the value and didn't reset it to 0.001 again.

**Comment by hoontee** [comment]
Not reproducible in master (fixed by #41199)
